src/setup.ts

```diff
@@ -1,93 +1,53 @@
-import { Buffer } from "buffer";
-import { ApplicationCommand, InteractionHandler } from "./types";
+import { Routes, RouteBases } from "discord-api-types/v10";
+import { RESTGetAPIApplicationCommandsResult } from "discord-api-types/v10";
 
-const btoa = (value: string) => Buffer.from(value, "binary").toString("base64");
-
-const getAuthorizationCode = async (headers: any) => {
-  headers["Content-Type"] = "application/x-www-form-urlencoded";
-
-  const request = new Request("https://discord.com/api/oauth2/token", {
-    method: "POST",
-    body: new URLSearchParams({
-      grant_type: "client_credentials",
-      scope: "applications.commands.update",
-    }).toString(),
-    headers: headers,
-  });
-
-  const response = await fetch(request);
-
-  if (!response.ok) throw new Error("Failed to request an Authorization code.");
-
-  try {
-    const data = await response.json();
-    return data.access_token;
-  } catch {
-    throw new Error("Failed to parse the Authorization code response.");
-  }
-};
+import type { Application, Command } from "./handler";
 
 const resolveCommandsEndpoint = (applicationId: string, guildId?: string): string => {
-  const url = `https://discord.com/api/v8/applications/${applicationId}`;
-
-  if (guildId) {
-    return `${url}/guilds/${guildId}/commands`;
-  }
-
-  return `${url}/commands`;
+  if (guildId !== undefined) return RouteBases.api + Routes.applicationGuildCommands(applicationId, guildId);
+  return RouteBases.api + Routes.applicationCommands(applicationId);
 };
 
-const deleteExistingCommands = async (applicationId: string, bearer: any, guildId?: string): Promise<void> => {
+const deleteExistingCommands = async (applicationId: string, applicationSecret: string, guildId?: string): Promise<void> => {
```

Comment:
```suggestion
const deleteExistingCommands = async (applicationId: string, botToken: string, guildId?: string): Promise<void> => {
```

As it's not anymore the `CLIENT_SECRET` used in here, we're using the Bot's token.